### PR TITLE
Fix configuration property name when group is empty

### PR DESCRIPTION
**Pull request by ericbottard**

### CODE REVIEW

**Comment by snicoll** [comment]
Can you please add a test?

**Comment by ericbottard** [comment]
will do

**Comment by ericbottard** [comment]
Added test

File: spring-boot-tools/spring-boot-configuration-metadata/src/test/resources/metadata/configuration-metadata-emptygroup.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "groups": [
+    {
+      "name": "",
```
Comment (snicoll) [inline spring-boot-tools/spring-boot-configuration-metadata/src/test/resources/metadata/configuration-metadata-emptygroup.json]: I am not sure that this is legit actually. It should be re-attached to the "root" group somehow. I am assuming that you end up in this situation with a plain `@ConfigurationProperties` annotation on your bean?

Comment (ericbottard) [inline spring-boot-tools/spring-boot-configuration-metadata/src/test/resources/metadata/configuration-metadata-emptygroup.json]: Yes, below is what we get generated for [CassandraSinkProperties](https://github.com/spring-cloud/spring-cloud-stream-modules/blob/26ba4dcd9d5b5da42fffcff2893f38502bfa8e87/cassandra-sink/src/main/java/org/springframework/cloud/stream/module/cassandra/sink/CassandraSinkProperties.java) for example.

```
{
  "groups": [
    {
      "name": "",
      "type": "org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkProperties",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkProperties"
    },
    {
      "name": "spring.cassandra",
      "type": "org.springframework.cloud.stream.module.cassandra.CassandraProperties",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties"
    }
  ],
  "properties": [
    {
      "name": "consistency-level",
      "type": "org.springframework.cassandra.core.ConsistencyLevel",
      "description": "The consistencyLevel option of WriteOptions.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkProperties"
    },
    {
      "name": "ingest-query",
      "type": "java.lang.String",
      "description": "The ingest Cassandra query.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkProperties"
    },
    {
      "name": "query-type",
      "type": "org.springframework.integration.cassandra.outbound.CassandraMessageHandler<T>$Type",
      "description": "The queryType for Cassandra Sink.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkProperties"
    },
    {
      "name": "retry-policy",
      "type": "org.springframework.cassandra.core.RetryPolicy",
      "description": "The retryPolicy option of WriteOptions.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkProperties"
    },
    {
      "name": "spring.cassandra.compression-type",
      "type": "org.springframework.cassandra.config.CompressionType",
      "description": "The compression to use for the transport.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties"
    },
    {
      "name": "spring.cassandra.contact-points",
      "type": "java.lang.String",
      "description": "The comma-delimited string of the hosts to connect to Cassandra.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties"
    },
    {
      "name": "spring.cassandra.create-keyspace",
      "type": "java.lang.Boolean",
      "description": "The flag to create (or not) keyspace on module startup.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties",
      "defaultValue": false
    },
    {
      "name": "spring.cassandra.entity-base-packages",
      "type": "java.lang.String[]",
      "description": "The base packages to scan for entities annotated with Table annotations.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties",
      "defaultValue": []
    },
    {
      "name": "spring.cassandra.init-script",
      "type": "org.springframework.core.io.Resource",
      "description": "The resource with CQL scripts (delimited by ';') to initialize keyspace schema.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties"
    },
    {
      "name": "spring.cassandra.keyspace",
      "type": "java.lang.String",
      "description": "The keyspace name to connect to.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties"
    },
    {
      "name": "spring.cassandra.metrics-enabled",
      "type": "java.lang.Boolean",
      "description": "Enable/disable metrics collection for the created cluster.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties"
    },
    {
      "name": "spring.cassandra.password",
      "type": "java.lang.String",
      "description": "The password for connection.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties"
    },
    {
      "name": "spring.cassandra.port",
      "type": "java.lang.Integer",
      "description": "The port to use to connect to the Cassandra host.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties"
    },
    {
      "name": "spring.cassandra.schema-action",
      "type": "org.springframework.data.cassandra.config.SchemaAction",
      "description": "The schema action to perform.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties"
    },
    {
      "name": "spring.cassandra.username",
      "type": "java.lang.String",
      "description": "The username for connection.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.CassandraProperties"
    },
    {
      "name": "statement-expression",
      "type": "org.springframework.expression.Expression",
      "description": "The expression in Cassandra query DSL style.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkProperties"
    },
    {
      "name": "ttl",
      "type": "java.lang.Integer",
      "description": "The time-to-live option of WriteOptions.",
      "sourceType": "org.springframework.cloud.stream.module.cassandra.sink.CassandraSinkProperties",
      "defaultValue": 0
    }
  ],
  "hints": []
}
```

**Comment by snicoll** [comment]
Merged, thanks for the PR!

**Pull request closed without merging**